cargo-dist/src/main.rs

```diff
@@ -516,3 +519,57 @@ fn cmd_manifest_schema(
     }
     Ok(())
 }
+
+async fn cmd_update(_config: &Cli, args: &cli::UpdateArgs) -> Result<(), miette::ErrReport> {
+    let mut updater = AxoUpdater::new_for("cargo-dist");
+    // Do we want to treat this as an error?
+    // Or do we want to sniff if this was a Homebrew installation?
+    if updater.load_receipt().is_err() {
+        println!("Unable to load install receipt to check for updates.");
+        println!("If you installed this via `brew`, please `brew upgrade cargo-dist`!");
+        return Ok(());
+    }
+
+    if !updater.check_receipt_is_for_this_executable()? {
```

Comment:
why only handle the boolean failure and not the error failure? why are these materially different?

Reply:
The boolean isn't an error, it's an answer to the question: is this receipt for this executable? The error is if we encountered an error trying to calculate an answer to that question.